app/repositories/product_repository.rb

```diff
@@ -0,0 +1,146 @@
+require 'date'
+
+# repositories offer a business abstraction to access data, models should only be used inside them
+class ProductRepository
+  # @param id [Integer]
+  # @return [BookDto, ImageDto, VideoDto]
+  def get_product(id)
```

Comment:
Can this function use `fetch_details`?

Reply:
It can but it would make 2 useless SQL query (as just one kind will be used). I don't have string opinion between a "useless" performance diff and a trivial code duplication.
In any way, I think it's just a not important detail as it's private code and could be easily changed (no API change)